server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java

```diff
@@ -87,6 +88,24 @@ default Collection<RestHandler> getRestHandlers(
         return Collections.emptyList();
     }
 
+    /**
+    * Rest handlers added by this plugin with ThreadContext.
+    */
+    default Collection<RestHandler> getRestHandlers(
+        Settings settings,
+        NamedWriteableRegistry namedWriteableRegistry,
+        RestController restController,
+        ClusterSettings clusterSettings,
+        IndexScopedSettings indexScopedSettings,
+        SettingsFilter settingsFilter,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<DiscoveryNodes> nodesInCluster,
+        Predicate<NodeFeature> clusterSupportsFeature,
+        ThreadContext context
```

Comment:
A general question for core/infra perhaps: are we good using ThreadContext to set transport headers like we do in this PR? I believe that an advantage is that it can be used in other places, as opposed to adding a specific SearchRequest flag that would be specific to search. I'd assume that the same problem around carrying around stacktraces without ever returning them to the user is common.